apps/css-workshop/src/components/Checkbox.astro

```diff
@@ -0,0 +1,71 @@
+---
+type Props = {
+  isLoading?: boolean;
+  indeterminate?: boolean;
+  checked?: boolean;
+  disabled?: boolean;
+  variant?: 'eyeball';
+  basic?: boolean;
+  status?: 'negative' | 'positive' | 'warning';
+} & astroHTML.JSX.HTMLAttributes;
+
+const { status, isLoading, indeterminate, disabled, variant, basic, ...props } = Astro.props;
+---
+
+{
+  basic && (
+    <input
+      class:list={[
+        'iui-checkbox',
+        {
+          'iui-loading': isLoading,
+          'iui-checkbox-visibility': variant?.includes('eyeball'),
+        },
+      ]}
+      type='checkbox'
+      name={indeterminate ? 'indeterminate' : undefined}
+      disabled={disabled}
+      {...props}
+    />
+  )
+}
+{
+  !basic && (
+    <label
+      class:list={[
+        'iui-checkbox-wrapper',
+        {
+          'iui-disabled': disabled,
+          'iui-loading': isLoading,
+          [`iui-${status}`]: !!status,
+        },
+      ]}
+      {...props}
+    >
+      <input
+        class:list={[
+          'iui-checkbox',
+          {
+            'iui-loading': isLoading,
+            'iui-checkbox-visibility': variant?.includes('eyeball'),
+          },
+        ]}
+        type='checkbox'
+        name={indeterminate ? 'indeterminate' : undefined}
```

Comment:
`name` should not be set here.

Reply:
![Screenshot 2024-04-05 141900](https://github.com/iTwin/iTwinUI/assets/148785417/8665e6f2-9234-40b5-927a-90c4d556166b)

The original script used `name` to add the `indeterminate` mask image so I did the same. 

The CSS selector is using the indeterminate pseudo class which I don't think is [directly assignable without Javascript](https://css-tricks.com/almanac/selectors/i/indeterminate/#aa-it-cannot-be-set-in-the-html). 
 
Using the name attribute seemed like a fairly simple way of doing it. But if there's better way to do it, I can do that instead.

Reply:
The [`name`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#name) attribute belongs to the individual checkboxes, and shouldn't be set at the component level.

`indeterminate` can totally be just be set as a regular attribute in HTML (even if it doesn't do anything by itself). The JavaScript code can then hook read the HTML attribute, and convert it to the `indeterminate` property.